src/main/java/com/karankumar/bookproject/backend/service/BookService.java

```diff
@@ -48,6 +53,10 @@ public Book findById(Long id) {
 
     public void save(Book book) {
         if (bookHasAuthorAndPredefinedShelf(book)) {
+        	if(bookHasPublisher(book)) {
```

Comment:
Any reason to have this under the `if (bookHasAuthorAndPredefinedShelf(book))` ? Since we have a separate publisher service so all the logic related to saving publisher should be in that service.

Reply:
Whats the strategy to handle if there is no publisher information? Its not mandatory like Author, right?

Reply:
Look at the new way of handling it